spring-boot-actuator/src/main/java/org/springframework/boot/actuate/info/InfoProvider.java

```diff
@@ -0,0 +1,17 @@
+package org.springframework.boot.actuate.info;
+
+import java.util.Map;
+
+/**
+ * information provider for the info endpoint
+ *
+ * @author Meang Akira Tanaka
+ */
+public interface InfoProvider {
+
+	/**
+	 * @return a collection of information
+	 */
+	Map<String, Object> provide();
```

Comment:
Hi Stephane

I can do that but I think I will break the interface when I change the
interface to return another type than Map no?

Or do you have any suggestions?

br

Meang


On Tue, Jul 14, 2015 at 9:21 PM, Stéphane Nicoll <notifications@github.com>
wrote:

> In
> spring-boot-actuator/src/main/java/org/springframework/boot/actuate/info/InfoProvider.java
> <https://github.com/spring-projects/spring-boot/pull/3492#discussion_r34607930>
> :
>
> > @@ -0,0 +1,17 @@
> > +package org.springframework.boot.actuate.info;
> > +
> > +import java.util.Map;
> > +
> > +/**
> > + * information provider for the info endpoint
> > + *
> > + * @author Meang Akira Tanaka
> > + */
> > +public interface InfoProvider {
> > +
> > +	/**
> > +	 * @return a collection of information
> > +	 */
> > +	Map<String, Object> provide();
>
> I'd prefer a more fine-grained API *à la* Health
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spring-projects/spring-boot/pull/3492/files#r34607930>
> .
>